server/tools/peertube-import-videos.ts

```diff
@@ -33,27 +34,64 @@ command
   .option('-U, --username <username>', 'Username')
   .option('-p, --password <token>', 'Password')
   .option('--target-url <targetUrl>', 'Video target URL')
+  .option('--tmpdir <tmpdir>', 'Working directory')
   .option('--since <since>', 'Publication date (inclusive) since which the videos can be imported (YYYY-MM-DD)', parseDate)
   .option('--until <until>', 'Publication date (inclusive) until which the videos can be imported (YYYY-MM-DD)', parseDate)
-  .option('-v, --verbose', 'Verbose mode')
+  .option('--first <first>', 'Process first n elements of returned playlist')
+  .option('--last <last>', 'Process last n elements of returned playlist')
+  .option('-v, --verbose [verbosity]', 'Verbosity, from 0 (only report errors) to 4 (for debugging), default is 2 (info)')
   .parse(process.argv)
 
 getServerCredentials(command)
   .then(({ url, username, password }) => {
     if (!program[ 'targetUrl' ]) {
-      console.error('--targetUrl field is required.')
+      log.error('--targetUrl field is required.')
 
       process.exit(-1)
     }
 
+    if (!program['tmpdir']) {
+      program[ 'tmpdir' ] = __dirname;
```

Comment:
Commander allows specifying the default value. Rather adding this kind of conditions, we'd probably rather do:
```js
command
  // …
  .option('--tmpdir <tmpdir>', 'Working directory', __dirname)
  // …
```

Reply:
True, I more or less hacked this change into the program to be able to mirror a channel which has been blocked for no good reason, then unblocked again. This is clear from many of the additions, including...